### DISCUSSION TITLE
Feature request: Skipped slides

### DISCUSSION THREAD

**Discussion by Maximvdw** [hakimel/reveal.js#3063] [Ideas]
**Use case**
You are creating a presentation and want to trim down on the amount of content. Currently you can use ```data-visibility="hidden"``` to hide the slides - but that also means that you have no way of using those slides in your presentation. The documentation proposes 'uncounted' slides for this use case - but this means that you should move the slides from their logical (2D) position to the back of your presentation. Instead I propose:

```data-visibility="skipped"```

- that will skip the slide during navigation
- that will not count the slide in the progress bar
- that can still be accessed from the slide overview

In essence it is very similar to hidden slides, but not in the 'overview'.

**Behaviour**
'skipped slides' should not be shown during normal navigation, they should not be counted in the total progress but unlike ```data-visibility="hidden"``` it should still be possible to select them in the overview.

**Comment by peter-lyons-kehl** [comment]
Do you use/need [vertical slides](https://revealjs.com/vertical-slides/)? If not, those can serve this purpose. You can hide the navigation arrows (`controls`) in [config](https://revealjs.com/config) as not to confuse/distract your audience.

If you use Markdown, you can split the content into several ["External" Markdown](https://revealjs.com/markdown/#external-markdown) files (whether on the same site/domain or different). Then you could comment out loading of specific files in your `index.html`, as you need for a particular presentation.

**Reply by Maximvdw** [reply]
Hi, I do need vertical slides. My use case is that I have a normal presentation (horizontal and vertical slides) but I want to trim down on the length so I "hide" slides that I do not want to include in my final presentation.

I do not want the progress bar to suddenly "jump" a few places (which would be the case if you skip a vertical slide) and I do not want to artificially split my already vertical slides in another section just to hide them.

At the end - when I do my final presentation, I want to still be able to see those pages in the overview so I can use them as a backup slide for questions or in case I speak faster than expected.

**Comment by Maximvdw** [comment]
FYI, this is how I currently solve it - but it would be great if Reveal.js officially supports this

```js
Reveal.on('slidechanged', (event) => {
  if (event.currentSlide.hasAttribute("data-visibility") && !Reveal.isOverview()) {
    const prevIndices = Reveal.getIndices(event.previousSlide);
    const currentIndices = Reveal.getIndices(event.currentSlide);
    if (prevIndices.h < currentIndices.h || (prevIndices.h === currentIndices.h && prevIndices.v < currentIndices.v)) {
      Reveal.next();
    } else if (prevIndices.h > currentIndices.h || (prevIndices.h === currentIndices.h && prevIndices.v > currentIndices.v)) {
      Reveal.prev();
    }
  }
});
```
^aka. If the slide is changed, and the current slide that it was changed to has the attribute "data-visibility" then we assume we want to skip it. In that case we use Reveal.next or prev depending on the direction we changed (and if we have a next/prev slide).

Final check is to make sure that we 'can' select the slide in the overview which is why we have `!Reveal.isOverview()`


NOTE: I wrote this an hour before a conference presentation so suffice it to say that I did not investigate if there was an API method for the long if statements, etc... it works - but its probably not optimal
